fast_api_als/routers/submit_lead.py

```diff
@@ -1,96 +1,133 @@
 import time
 import uuid
-import logging
+from logger import logger
 
 from datetime import datetime
 from fastapi import APIRouter
-from fastapi import Request, Depends
+from fastapi import Request, Depends, HTTPException
 from fastapi.security.api_key import APIKey
 from concurrent.futures import ThreadPoolExecutor, as_completed
 
-from fast_api_als.services.authenticate import get_api_key
-from fast_api_als.services.enrich.customer_info import get_contact_details
-from fast_api_als.services.enrich.demographic_data import get_customer_coordinate
-from fast_api_als.services.enrich_lead import get_enriched_lead_json
-from fast_api_als.services.new_verify_phone_and_email import new_verify_phone_and_email
-from fast_api_als.utils.adf import parse_xml, check_validation
-from fast_api_als.utils.calculate_lead_hash import calculate_lead_hash
-from fast_api_als.database.db_helper import db_helper_session
-from fast_api_als.services.ml_helper import conversion_to_ml_input, score_ml_input
-from fast_api_als.utils.quicksight_utils import create_quicksight_data
-from fast_api_als.quicksight.s3_helper import s3_helper_client
-from fast_api_als.utils.sqs_utils import sqs_helper_session
+from services.authenticate import get_api_key
+from services.enrich.customer_info import get_contact_details
+from services.enrich.demographic_data import get_customer_coordinate
+from services.enrich_lead import get_enriched_lead_json
+from services.new_verify_phone_and_email import new_verify_phone_and_email
+from utils.adf import parse_xml, check_validation
+from utils.calculate_lead_hash import calculate_lead_hash
+from database.db_helper import db_helper_session
+from services.ml_helper import conversion_to_ml_input, score_ml_input
+from utils.quicksight_utils import create_quicksight_data
+from quicksight.s3_helper import s3_helper_client
+from utils.sqs_utils import sqs_helper_session
 
 router = APIRouter()
 
 """
-Add proper logging and exception handling.
+Add proper logger and exception handling.
 
 keep in mind:
 You as a developer has to find how much time each part of code takes.
 you will get the idea about the part when you go through the code.
 """
 
+
 @router.post("/submit/")
 async def submit(file: Request, apikey: APIKey = Depends(get_api_key)):
+
     start = int(time.time() * 1000.0)
-    t1 = [int(time.time() * 1000.0)]
-    
     if not db_helper_session.verify_api_key(apikey):
-        # throw proper fastpi.HTTPException
-        pass
-    
+        logger.error("[Submit File]: API key not verified")
+        raise HTTPException(status_code=401, detail="API key not verified")
+
     body = await file.body()
     body = str(body, 'utf-8')
 
+    now = int(time.time() * 1000.0)
+    logger.info(
+        f'[Submit File]: file body parsed and converted to string in {now - start}ms')
```

Comment:
You shouldn't have `[Submit File]: ` in every log statement.

Reply:
How would I pinpoint to what task a particular log is from?

Using `[Task Name]` was my way of _tagging_ a log, is there a better way?

Reply:
Whenever you're seeing logs you're seeing a group of logs right, that is following a story.
At the start of your story, you can say `I'm executing this function`.
agree or have any comments?

Reply:
That makes sense for linear flows. 
Although for concurrent scenarios where multiple functions do similar tasks, I guess a tag could be helpful, say for example we have something like the following: 
```
async func1()
{ 
print('a completed')
print('b completed')
 }

async func2()
{
print('a completed')
print('b completed')
 }

func1()
func2()
```
here our logs may look like
```
a completed
a completed
b completed
b completed
```

adding tags here can help us understand the story better
```
[func1] a completed
[func2] a completed
[func2] b completed
[func1] b completed
```

Reply:
Yes, this comes under the special case.